main.tf

```diff
@@ -112,7 +119,8 @@ resource "aws_eks_cluster" "default" {
 
   depends_on = [
     aws_iam_role_policy_attachment.amazon_eks_cluster_policy,
-    aws_iam_role_policy_attachment.amazon_eks_service_policy
+    aws_iam_role_policy_attachment.amazon_eks_service_policy,
+    aws_cloudwatch_log_group.default
```

Comment:
@kskewes 
Can you explain in more details why we need to create `resource "aws_cloudwatch_log_group"` in the module?
Looks like the EKS cluster and `aws_cloudwatch_log_group` are unrelated, meaning there is no connection b/w/ them except `name              = "/aws/eks/${module.label.id}/cluster"`.

Also, from 

```
depends_on = [
    aws_iam_role_policy_attachment.amazon_eks_cluster_policy,
    aws_iam_role_policy_attachment.amazon_eks_service_policy
    aws_iam_role_policy_attachment.amazon_eks_service_policy,
    aws_cloudwatch_log_group.default
```

it looks like you want to create `aws_cloudwatch_log_group` first anyway, so why it could not be created outside of the cluster when needed?

Thanks again for your contributions.

Reply:
If you add `audit` or other log types to the module then on cluster creation AWS will create the log group for you.
It creates the log group with an unlimited retention period by default.
If you then want to update the retention period you need to `terraform import` the log group and adjust.
So in our accounts we enforce creating the log group first via a `depends_on`.
I apologise I haven't validated the `depends_on` when `count == 0` per your review comment above. :pray: 
That validation should really block this PR and if it's not possible then I think we should close the PR.

Reply:
Note that modules can't use `depends_on` (yet), else we would put it there.